Microsoft.Identity.Web/WebApiStartupHelpers.cs

```diff
@@ -87,7 +93,12 @@ public static IServiceCollection AddProtectedApiCallsWebApis(this IServiceCollec
                         context.Success();
 
                         // Todo : rather use options.SaveToken?
-                        (context.Principal.Identity as ClaimsIdentity).AddClaim(new Claim("jwt", (context.SecurityToken as JwtSecurityToken).RawData));
+                        JwtSecurityToken jwtSecurityToken = context.SecurityToken as JwtSecurityToken;
```

Comment:
@jmprieur, @bgavrilMS Currently, AAD deflates JWE and the JwtSecurityTokenHandler does not support deflating and JWE's from AAD will fault with an error indicating a 'Read' failure. The JsonWebTokenHandler does support JWE's with deflate so there is a way to make this work. Let me know if you want me to issue a PR.

Reply:
Yes, please, @brentschmaltz . 
Would you use JsonWebTokenHandler? 
Or do we want to wait for Wilson 5.5.0 ? (which if I understand correctly will support JWE ?)

Reply:
@jmprieur JsonWebTokenHandler would be used. System.IdentityModel.Tokens.Jwt does support JWE, just not the optional 'deflate'. In 5.5.0, it will be supported.

Reply:
Thanks @brentschmaltz